datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -602,15 +622,38 @@ impl BufferedBatch {
             + mem::size_of::<Range<usize>>()
             + mem::size_of::<usize>();
 
+        let num_rows = batch.num_rows();
         BufferedBatch {
             batch,
             range,
             join_arrays,
             null_joined: vec![],
             size_estimation,
             join_filter_failed_idxs: HashSet::new(),
+            num_rows,
+            spill_file: None,
         }
     }
+
+    fn spill_to_disk(
+        &mut self,
+        path: RefCountedTempFile,
+        buffered_schema: SchemaRef,
+        batch_size: usize,
+    ) -> Result<()> {
+        let batch = std::mem::replace(
+            &mut self.batch,
+            RecordBatch::new_empty(Arc::clone(&buffered_schema)),
+        );
+        let _ = spill_record_batch_by_size(
+            batch,
+            path.path().into(),
+            buffered_schema,
+            batch_size,
+        );
+        self.spill_file = Some(path);
```

Comment:
I couldn't quite tell -- is it guaranteed that `self.spill_file` is None here. Is it possible to spill more than once? And if so, would this spill file get lost ?

Reply:
The batch can be spilled only once to the disk. I'm not sure if there is a necessity of spilling multiple times. But even if so the spill file will be pointing to the most recent valid spill file